cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -837,6 +834,54 @@ private void setOutputsIfNeeded(Job job) {
     }
   }
 
+  private void fixOutputPermissions(Job job, Map<String, ProvidedOutput> outputs) {
+    Set<String> outputsWithUmask = new HashSet<>();
+    Set<String> outputUmasks = new HashSet<>();
+    for (Map.Entry<String, ProvidedOutput> entry : outputs.entrySet()) {
+      String umask = entry.getValue().getOutputFormatConfiguration().get("fs.permissions.umask-mode");
+      if (umask != null) {
+        outputsWithUmask.add(entry.getKey());
+        outputUmasks.add(umask);
+      }
+    }
+    boolean allOutputsHaveUmask = outputsWithUmask.size() == outputs.size();
+    boolean allOutputsAgree = outputUmasks.size() == 1;
+    boolean jobConfHasUmask = ConfigurationUtil.isProgrammaticProperty(
+      job.getConfiguration().getPropertySources("fs.permissions.umask-mode"));
+    String jobConfUmask = job.getConfiguration().get("fs.permissions.umask-mode");
+
+    boolean mustFixUmasks = false;
+    if (jobConfHasUmask) {
+      // case 1: job conf has a programmatic umask. It prevails.
+      mustFixUmasks = !outputsWithUmask.isEmpty();
```

Comment:
Oh my purpose was that I only wanted to log if mustFixUmasks==true. Fixed that now.